src/psyclone/psyir/nodes/psy_data_node.py

```diff
@@ -756,10 +757,16 @@ def gen_type_bound_call(typename, methodname, argument_list=None,
                 f"lowering but '{self}' is not.")
 
         self.generate_symbols(routine_schedule.symbol_table)
-
         module_name = self._module_name
         if module_name is None:
-            module_name = routine_schedule.name
+            container = routine_schedule.ancestor(Container)
+            # See if there current code is inside of a module. If so,
```

Comment:
"If the current code is inside a module use the module name, otherwise ...."

Reply:
Done.